tests/test_google_ads_interrupted_sync_remove_stream.py

```diff
@@ -0,0 +1,238 @@
+import os
+from datetime import datetime as dt
+from datetime import timedelta
+
+from tap_tester import menagerie, connections, runner
+
+from base import GoogleAdsBase
+
+
+class InterruptedSyncRemoveStreamTest(GoogleAdsBase):
+    """Test tap's ability to recover from an interrupted sync"""
+
+    @staticmethod
+    def name():
+        return "tt_google_ads_interruption_remove"
+
+    def get_properties(self, original: bool = True):
+        """Configurable properties, with a switch to override the 'start_date' property"""
+        return_value = {
+            'start_date':   '2022-01-22T00:00:00Z',
+            'user_id':      'not used?', # TODO ?
+            'customer_ids': ','.join(self.get_customer_ids()),
+            'login_customer_ids': [{"customerId": os.getenv('TAP_GOOGLE_ADS_CUSTOMER_ID'),
+                                    "loginCustomerId": os.getenv('TAP_GOOGLE_ADS_LOGIN_CUSTOMER_ID'),}],
+        }
+
+        # TODO_TDL-17911 Add a test around conversion_window_days
+        if original:
+            return return_value
+
+        self.start_date = return_value['start_date']
+        return return_value
+
+
+    def test_run(self):
+        """
+        Scenario: A sync job is interrupted. The state is saved with `currently_syncing`.
+                  The next sync job kicks off and the tap picks back up on that `currently_syncing` stream.
+
+        Expected State Structure:
+            state = {'currently_syncing': ('<stream-name2>', '<customer-id>'),
+            'bookmarks': {
+                '<stream-name1>': {'<customer-id>': {'<replication-key>': <completed-bookmark-value>}},
+                '<stream-name2>': {'<customer-id>': {'<replication-key>': <incomplete-bookmark-value>}},
+
+        Test Cases:
+         - Verify behavior is consistent when a stream is removed from selected list between initial and resuming sync
+         - Verify an interrupted sync can resume based on the `currently_syncing` and stream level bookmark value
+         - Verify only records with replication-key values greater than or equal to the stream level bookmark are replicated on the resuming sync for the interrupted stream
+         - Verify the yet-to-be-synced streams are replicated following the interrupted stream in the resuming sync. All yet-to-be-synced streams must replicate before streams that were already synced.
```

Comment:
Same comment as currently syncing test. We are not verifying that the removed stream does not get synced.